addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/thermostat/NATherm1Handler.java

```diff
@@ -168,24 +262,40 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                             spTemp = new BigDecimal(command.toString()).setScale(1, RoundingMode.HALF_UP);
                         }
                         if (spTemp != null) {
-                            // Switch the thermostat to manual mode on the desired setpoint for given duration
-                            Calendar cal = Calendar.getInstance();
-                            cal.add(Calendar.MINUTE, getSetPointDefaultDuration());
-                            getBridgeHandler().getThermostatApi().setthermpoint(getParentId(), getId(), "manual",
-                                    (int) (cal.getTimeInMillis() / 1000), spTemp.floatValue());
-                            updateState(channelUID, new QuantityType<Temperature>(spTemp, API_TEMPERATURE_UNIT));
-                            requestParentRefresh();
+                            pushSetpointUpdate(CHANNEL_SETPOINT_MODE_MANUAL, getSetpointEndTime(), spTemp.floatValue());
                         }
+
                         break;
                     }
+                    case CHANNEL_PLANNING: {
+                        getBridgeHandler().getThermostatApi().switchschedule(getParentId(), getId(),
+                                command.toString());
+                        updateState(channelUID, new StringType(command.toString()));
+                        invalidateParentCacheAndRefresh();
+                    }
                 }
             } catch (Exception e) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
             }
         }
     }
 
+    private void pushSetpointUpdate(String target_mode, Integer setpointEndtime, Float setpointTemp) {
+        getBridgeHandler().getThermostatApi().setthermpoint(getParentId(), getId(), target_mode, setpointEndtime,
+                setpointTemp);
+        invalidateParentCacheAndRefresh();
+    }
+
+    private int getSetpointEndTime() {
+        Calendar cal = Calendar.getInstance();
```

Comment:
See above, my remark about time calculations

Reply:
Sure but it works, and you said it would not be blocking. I would prefer avoid diving in UTC, Time zones, durations :) for a tested piece of code that is ok.